airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py

```diff
@@ -417,6 +432,12 @@ def interpret_response(self, response_or_exception: Optional[Union[requests.Resp
             return super().interpret_response(response_or_exception)
         should_retry = self.stream.should_retry(response_or_exception)  # type: ignore # noqa
         if should_retry:
+            if response_or_exception.status_code == 429:  # type: ignore[union-attr]
```

Comment:
we should not be ignoring these. If this is of type `Exception` this would crash? We probably need to type check before we attempt to get the `status_code`.

Reply:
added type check before getting `should_retry`